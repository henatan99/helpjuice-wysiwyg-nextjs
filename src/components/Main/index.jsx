import Image from "next/image"
import Editor from "../Editor"
import ToolBar from "../ToolBar"

const Main = () => {
    return (
        <div style={{width: '100%', display: 'flex', flexDirection: 'column', alignItems: 'center', justifyContent: 'center'}}>
            
            <div style={{width: '50%'}}>
                <div>
                    <ToolBar />
                </div>
                <h1>
                    Front-end Developer Test Project
                </h1>
                <p>
                    Your goal is to make a page that looks exactly like this one, and has the ability to create 
                    H1 text simply by typing text, and hitting enter.
                </p>
                <div style={{width: '100%'}}>
                    <Editor />
                </div>
            </div>
        </div>
        
    )
}

export default Main