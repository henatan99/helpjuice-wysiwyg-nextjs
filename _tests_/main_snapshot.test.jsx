import { render } from '@testing-library/react'
import Main from '../src/components/Main'

it('renders main unchanged', () => {
  const { container } = render(<Main />)
  expect(container).toMatchSnapshot()
})