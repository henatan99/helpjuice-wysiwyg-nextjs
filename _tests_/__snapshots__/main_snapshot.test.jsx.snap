// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders main unchanged 1`] = `
<div>
  <div
    style="width: 100%; display: flex; flex-direction: column; align-items: center; justify-content: center;"
  >
    <div
      style="width: 50%;"
    >
      <div>
        <div
          style="display: flex; justify-content: space-between; align-items: center; padding: 0.3rem; border: 0.2px solid #e5e7eb; border-radius: 0.3rem; text-align: center; box-shadow: 0 0.25px 0.89px 0.125px rgb(64 60 67 / 16%);"
        >
          <div
            style="display: flex; align-items: center;"
          >
            <span
              style="margin-right: 0.8rem; text-align: center; color: rgb(157, 163, 174); display: flex;"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2723%27%20height=%2727%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="P"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </span>
            <span
              style="margin-right: 0.6rem; font-size: 11px; text-align: center; color: rgb(157, 163, 174); display: flex;"
            >
              |
            </span>
            <span
              style="margin-right: 0.3rem; text-align: center; color: rgb(157, 163, 174); display: flex;"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2713%27%20height=%2713%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="clock"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </span>
            <span
              style="margin-right: 0.8rem; font-size: 14px; text-align: center; color: rgb(157, 163, 174); display: flex;"
            >
              0min
            </span>
            <span
              style="margin-right: 0.7rem; font-size: 11px; text-align: center; color: rgb(157, 163, 174); display: flex;"
            >
              |
            </span>
            <span
              style="margin-right: 0.8rem; text-align: center; color: rgb(157, 163, 174); display: flex;"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2721%27%20height=%2721%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="rabbit"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </span>
            <span
              style="margin-right: 0.7rem; font-size: 11px; text-align: center; color: rgb(157, 163, 174); display: flex;"
            >
              |
            </span>
            <span
              style="margin-right: 0.5rem; text-align: center; color: rgb(157, 163, 174); display: flex;"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%278%27%20height=%278%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="arrow"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </span>
            <span
              style="margin-right: 1rem; font-size: 14px; text-align: center; color: rgb(157, 163, 174); display: flex;"
            >
              0
            </span>
          </div>
          <div
            style="display: flex; align-items: center;"
          >
            <span
              style="margin-right: 0.6rem; text-align: center; color: rgb(157, 163, 174); display: flex;"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2714%27%20height=%2714%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="check-mark"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </span>
            <span
              style="margin-right: 0.5rem; text-align: center; color: rgb(157, 163, 174); display: flex;"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2714%27%20height=%2710%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="cloud"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </span>
            <span
              style="text-align: center; color: rgb(157, 163, 174); display: flex;"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2714%27%20height=%2714%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="more"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </span>
          </div>
        </div>
      </div>
      <h1
        style="font-size: 2.4rem; margin-block-end: 0;"
      >
        Front-end developer test project
      </h1>
      <hr
        style="color: rgb(229, 231, 235); background: rgb(229, 231, 235); border: 0.25px solid #e5e7eb;"
      />
      <p
        style="color: rgb(99, 106, 118); line-height: 1.5rem; font-size: 15px;"
      >
        Your goal is to make a page that looks exactly like this one, and has the ability to create H1 text simply by typing text, and hitting enter.
      </p>
      <div
        style="width: 100%; margin-top: 3.5rem;"
      >
        <div
          class="sc-bczRLJ cHvIj"
        >
          <div
            style="font-size: 1rem; margin-bottom: 1rem; line-height: 1.8rem; font-family: Arial, Helvetica, sans-serif; position: relative;"
          />
          <textarea
            placeholder="Type / for blocks, @ to link docs or people"
            rows="20"
            style="padding: 0px 0px 0px 1%; border-width: 0px; width: 101%; max-width: unset; color: rgb(99, 106, 118); margin: -1%; outline: none; font-family: inherit;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
